xbmc/Application.cpp

```diff
@@ -3071,6 +3071,7 @@ PlayBackRet CApplication::PlayStack(const CFileItem& item, bool bRestart)
     movieList[selectedFile - 1]->m_lStartOffset = startoffset > 0 ? STARTOFFSET_RESUME : 0;
     movieList[selectedFile - 1]->SetProperty("stackFileItemToUpdate", true);
     *m_stackFileItemToUpdate = item;
+    // @todo this could be an issue, because restart=false does clear the stack/movieList. correct it or document why this is correct
```

Comment:
Is it intneded to keep this?

Reply:
It's correct that the m_stackFileItemToUpdate object maintains a reference to the overarching stack item that needs to be updated when saving a bookmark. But there's also the current item's boolean property "stackFileItemToUpdate" that tells you whether that is needed or not. As you see in the next line, there is a recursive call to PlayFile to play the selected part of the stack. When saving the bookmark, it will use the part's property "stackFileItemToUpdate" and if set (true) then it knows to store the bookmark with the stack instead of the part. 

Now what you could try in this PR, is to query the same file item's (GetCurrentFileItem) property (GetProperty("stackFileItemToUpdate")) and if set to true, you know that in fact you were playing a stack part and to resume, you need to call PlayStack(*m_stackFileItemToUpdate, true).

For the other kind of stacks (see case 2 below) you need to check m_currentStack size() and if that's > 0 then you *could* simply call PlayFile(GetCurrentFileItem(), "", true)... - to be confirmed.

Reply:
disregard the above comment. Stacking was a terrible mess and I've cleaned that up in separate PR.

Reply:
Thank you, i will remove the @todo